db/products.js

```diff
@@ -0,0 +1,102 @@
+/*jshint esversion:6*/
+
+class Products {
+  constructor(){
+    this.id = 10001;
+    this._collection = [];
+  }
+
+  getAllProducts(){
+    return this._collection;
+  }
+
+  addProduct(newProduct){
+    for(var i=0; i<this._collection.length; i++){
+      if(this._collection[i].name === newProduct.name){
+        throw new Error ('Product name already exists');
+        }
+      }
+    this._collection.push(newProduct);
```

Comment:
You should probably create a new product object here and move over the properties ( name, price, inventory ) from newProduct on to the new product object. That way, the creation of the new product isn't outside this class.